### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by jiri-mikaus**

### CODE REVIEW

File: s3_test.go [hunk 1]
```diff
@@ -126,7 +124,6 @@ func uploadResolverFiles(dataFiles []string) {
 }
 
 func TestCorrectClientWithS3(t *testing.T) {
-	go minio.Main(args)
```
Comment (jiri-mikaus) [inline s3_test.go:129]: Minio didn't want to start so I added compose and gh action service. I also tried start it with our certificates as was starting this one but there was issue with missing SAN and I don't want to spend time on generating new certs. From my point of view it's ok without HTTPS in tests.

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        severity: warning
+        arguments:
+          - maxLitCount: "3"
+            allowStrs: '""'
+            allowInts: "0,1,2,3"
+            allowFloats: "0.0,0.,1.0,1.,2.0,2."
+
+issues:
+# run linter only on commits after the rev
+  new-from-rev: ed3a1c1c84ac975b595d199b15209c3401515bff
```
Comment (jiri-mikaus) [inline .golangci.yml:120]: Too many lint issues so I enabled it only for new changes.

File: server_test.go [hunk 3]
```diff
@@ -319,105 +321,105 @@ func TestCorrectClientCached(t *testing.T) {
...
-		[]string{"SRV_BIND_PORT", bindPort},
-		[]string{"SRV_BIND_HOST", "localhost"},
+		{"SRV_CA_CERT_PEM_BASE64", caCertBase64},
+		{"SRV_SERVER_CERT_PEM_BASE64", serverCertBase64},
+		{"SRV_SERVER_KEY_PEM_BASE64", serverKeyBase64},
+		{"SRV_BIND_PORT", bindPort},
+		{"SRV_BIND_HOST", "localhost"},
 	}
-	interaction(t, "unknown-client", []string{}, []string{"SSL_ERROR_BAD_CERT_ALERT", "alert bad certificate"}, "", props)
+	interaction(t, "unknown-client", []string{}, []string{"TLS alert, unknown CA (560)", "alert unknown ca"}, "", props)
```
Comment (jiri-mikaus) [inline server_test.go:411]: This is caused by `curl` changed output. It was quite detective work but I tried old fedora and it matched original error message so I'm confident it's ok. Also commend in `Dockerfile:7` added some confidence (it failed on new versions even then).

Comment (rattuscz) [reply server_test.go:411]: 🥇

**Review by rattuscz — APPROVED** [review]
LGTM

File: config/settings.go [hunk 4]
```diff
@@ -151,13 +152,13 @@ func LoadSettings() Settings {
...
 	if len(settings.CRL_PEM_FILE) > 0 {
-		crlBytes, err = ioutil.ReadFile(settings.CRL_PEM_FILE)
+		crlBytes, err = os.ReadFile(settings.CRL_PEM_FILE)
 		if err != nil {
 			log.Fatal(MSG00023, err)
 		}
 	}
 	if len(crlBytes) > 1 {
-		settings.crl, err = x509.ParseCRL(crlBytes)
+		settings.CRL, err = x509.ParseRevocationList(crlBytes)
```
[the remark this answers is not in the document]
Comment (jiri-mikaus) [reply config/settings.go:161]: In following PR I added support for PEM back and decoding it to DER before `ParseRevocationList` call. So it's fully backward compatible.

**Pull request closed without merging**